editor/src/clj/editor/properties_view.clj

```diff
@@ -212,6 +214,49 @@
                         ctrls))
     box))
 
+(def label-drag-position (atom nil))
+(def label-drag-op-seq (atom nil))
+
+(defn- handle-label-drag-event! [property-fn drag-update-fn update-ui-fn ^MouseDragEvent event]
+  (.consume event)
+  (let [property (property-fn)
+        {:keys [key node-ids edit-type]} (property-fn)
+        {:keys [precision from-type to-type]} edit-type
+        [x y] [(.getX event) (.getY event)]
+        [prev-x prev-y] @label-drag-position
+        delta-x (- x prev-x)
+        delta-y (- prev-y y)
+        max-delta (if (> (abs delta-x) (abs delta-y)) delta-x delta-y)
+        update-val (cond-> (or precision 1.0)
+                     (.isShiftDown event) (* 10.0)
+                     (.isControlDown event) (* 0.1)
+                     (neg? max-delta) -)]
+    (reset! label-drag-position [x y])
+    (when (> (abs max-delta) 1)
```

Comment:
This check avoids some quirky updates, mainly because we support gestures on both axises.